api/src/main/java/org/pmiops/workbench/institution/InstitutionEmailDomainMapper.java

```diff
@@ -0,0 +1,33 @@
+package org.pmiops.workbench.institution;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.mapstruct.Mapper;
+import org.pmiops.workbench.db.model.DbInstitution;
+import org.pmiops.workbench.db.model.DbInstitutionEmailDomain;
+import org.pmiops.workbench.model.Institution;
+import org.pmiops.workbench.utils.mappers.MapStructConfig;
+
+@Mapper(config = MapStructConfig.class)
+public interface InstitutionEmailDomainMapper {
+  default Set<DbInstitutionEmailDomain> modelToDb(
+      final Institution modelInstitution, final DbInstitution dbInstitution) {
+    return Optional.ofNullable(modelInstitution.getEmailDomains()).orElse(Collections.emptyList())
+        .stream()
+        .distinct()
+        .map(
+            domain ->
+                new DbInstitutionEmailDomain().setEmailDomain(domain).setInstitution(dbInstitution))
+        .collect(Collectors.toSet());
+  }
+
+  default List<String> dbToModel(final Set<DbInstitutionEmailDomain> dbDomains) {
```

Comment:
As above, you should just be able to clue in MapStruct now to convert the `DbInstitutionEmailDomain` to a `String` and it will write your collection mapping for you.

please name this something like `dbDomainToDomainNameList()`.

Also, why a list and not a set?

Reply:
The model has a (sorted) list

Reply:
got this to work returning a SortedSet.  Thanks.